src/integrated/land.jl

```diff
@@ -0,0 +1,669 @@
+export LandModel
+"""
+    struct LandModel{
+        FT,
+        MM <: Soil.Biogeochemistry.SoilCO2Model{FT},
+        SM <: Soil.EnergyHydrology{FT},
+        VM <: Canopy.CanopyModel{FT},
+        SnM <: Snow.SnowModel{FT},
+    } <: AbstractLandModel{FT}
+        "The soil microbe model to be used"
+        soilco2::MM
+        "The soil model to be used"
+        soil::SM
+        "The canopy model to be used"
+        canopy::VM
+        "The snow model to be used"
+        snow::SnM
+    end
+
+A concrete type of land model used for simulating systems with 
+soil, canopy, snow, soilco2.
+$(DocStringExtensions.FIELDS)
+"""
+struct LandModel{
+    FT,
+    MM <: Soil.Biogeochemistry.SoilCO2Model{FT},
+    SM <: Soil.EnergyHydrology{FT},
+    VM <: Canopy.CanopyModel{FT},
+    SnM <: Snow.SnowModel{FT},
+} <: AbstractLandModel{FT}
+    "The soil microbe model to be used"
+    soilco2::MM
+    "The soil model to be used"
+    soil::SM
+    "The canopy model to be used"
+    canopy::VM
+    "The snow model to be used"
+    snow::SnM
+end
+
+
+
+"""
+    LandModel{FT}(;
+        soilco2_type::Type{MM},
+        soilco2_args::NamedTuple = (;),
+        land_args::NamedTuple = (;),
+        soil_model_type::Type{SM},
+        soil_args::NamedTuple = (;),
+        canopy_component_types::NamedTuple = (;),
+        canopy_component_args::NamedTuple = (;),
+        canopy_model_args::NamedTuple = (;),
+        snow_model_type::Type{SnM},
+        snow_args::NamedTuple = (;),
+        ) where {
+            FT,
+            SM <: Soil.EnergyHydrology{FT},
+            MM <: Soil.Biogeochemistry.SoilCO2Model{FT},
+            SnM <: Snow.SnowModel{FT}
+            }
+
+A constructor for the `LandModel`, which takes in the concrete model
+type and required arguments for each component, constructs those models,
+and constructs the `LandModel` from them.
+
+Each component model is constructed with everything it needs to be stepped
+forward in time, including boundary conditions, source terms, and interaction
+terms.
+"""
+function LandModel{FT}(;
+    soilco2_type::Type{MM},
```

Comment:
This is more of a style choice, but I think it would be nice to unify the order of component models across the `LandModel` struct, the arguments to this function, and the model creation inside this function.

e.g. move the `land_args` to the first position in this function, then within the function extract land args first, then set up soil co2 model, then soil model, then canopy, then snow

Reply:
the order could be alphabetical if that's easier to keep track of

Reply:
Sounds good, thank you!